Cartesian_kernel/include/CGAL/Cartesian/function_objects.h

```diff
@@ -4304,6 +4306,36 @@ namespace CartesianKernelFunctors {
         ? result_type(ON_ORIENTED_BOUNDARY)
         : opposite(ot);
     }
+
+    result_type
+    operator()(const Segment_2& s, const Triangle_2& t) const
+    {
+      typename K::Construct_source_2 source;
+      typename K::Construct_target_2 target;
+      const Point_2 a = source(s);
+      const Point_2 b = target(s);
+      const FT dX = b.x() - a.x();
+      const FT dY = b.y() - a.y();
+
+      const Point_2& p0 = t[0];
+      const Point_2& p1 = t[1];
+      const Point_2& p2 = t[2];
+      const FT R0 = p0.x() * p0.x() + p0.y() * p0.y();
+      const FT R1 = p1.x() * p1.x() + p1.y() * p1.y();
+      const FT R2 = p2.x() * p2.x() + p2.y() * p2.y();
+      const FT denominator = (p1.x() - p0.x()) * (p2.y() - p0.y()) +
+                             (p0.x() - p2.x()) * (p1.y() - p0.y());
+      const FT det = 2 * denominator * (a.x() * dY - a.y() * dX)
```

Comment:
Why don't we call the function `determinant()`?

Reply:
Because this `det` (maybe I should change its name) is a simplified version of the full definition of this orientation test (`determinant(a, b, circumcenter(p0, p1, p2))`)